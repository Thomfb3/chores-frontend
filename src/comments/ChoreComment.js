import React, { useContext } from "react";
import findUserInTeam from "../helpers/findUserInTeam";
import { determineProfilePic } from "../helpers/determineProfilePic";
import { formatLongDate, formatTime } from "../helpers/formatDate";
import UserContext from "../auth/UserContext";
import Avatar from '@mui/material/Avatar';


function ChoreComment({ user, comment, date }) {
    const { currentUser, currentTeamUsers } = useContext(UserContext);
    const choreCommentUser = findUserInTeam(user, currentTeamUsers);

    return (
        <div className="ChoreComment">
            <div className="ChoreComment__details">
                <div className="ChoreComment__image">
                    {choreCommentUser.profileImage &&
                        <Avatar
                            className="Profile__avatar--component"
                            alt={determineProfilePic(choreCommentUser)}
                            src={determineProfilePic(choreCommentUser)}
                            sx={{ width: 45, height: 45 }}
                        >{choreCommentUser.username.charAt(0)}</Avatar>
                    }
                </div>
                <div className="ChoreComment__info">
                    <p className="ChoreComment__name">{choreCommentUser.username}</p>
                    <p className="ChoreComment__date">{`${formatLongDate(date)} @ ${formatTime(date)}`}</p>
                    <div className="ChoreComment__comment">
                        <p>{comment}</p>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default ChoreComment