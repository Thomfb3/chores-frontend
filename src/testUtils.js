import React from "react";
import UserContext from "./auth/UserContext";
import ChoreCommentContext from "./comments/ChoreCommentContext";

const demoUser = {
  _id: "userid1",
  username: "testuser1",
  firstName: "testfirst1",
  lastName: "testlast1",
  profileImage: "defaultProfile.jpg",
  allTimePoints: 1000,
  currentPoints: 200,
  role: "admin"
};

const testTeam = {
  _id: "testTeam",
  name: "testTeam"
};

const testTeamUsers = [
  {
    _id: "userid1",
    username: "testuser1",
    firstName: "testfirst1",
    lastName: "testlast1",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "admin"
  },
  {
    _id: "userid2",
    username: "testuser2",
    firstName: "testfirst2",
    lastName: "testlast2",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid3",
    username: "testuser3",
    firstName: "testfirst3",
    lastName: "testlast3",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid4",
    username: "testuser4",
    firstName: "testfirst4",
    lastName: "testlast4",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  }
];

let choreComments = [];

const UserProvider =
  ({ children,
    currentUser = demoUser,
    currentTeam = testTeam,
    currentTeamUsers = testTeamUsers,
    isAdmin = false
  }) => (
      <UserContext.Provider value={{ currentUser, currentTeam, currentTeamUsers }}>
        {children}
      </UserContext.Provider>
    );


const CommentProvider =
  ({ children,
    currentChoreComments = choreComments
  }) => (
      <ChoreCommentContext.Provider value={{ currentChoreComments }}>
        {children}
      </ChoreCommentContext.Provider>
    );

export { UserProvider, CommentProvider };

