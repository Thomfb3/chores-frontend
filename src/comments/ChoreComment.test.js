import React from "react";
import { render } from "@testing-library/react";
import ChoreComment from "./ChoreComment";
import { UserProvider } from "../testUtils";

it("matches snapshot", function () {
  
    const { asFragment } = render(
            <UserProvider>
                <ChoreComment
                    id="userid1"
                    key="userid1"
                    user="userid1"
                    comment="I'm commenting on the chore"
                    date="2021-11-05T02:24:00.000+00:00"
                />
            </UserProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
});
