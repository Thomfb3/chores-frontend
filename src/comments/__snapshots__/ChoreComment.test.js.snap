// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ChoreComment"
  >
    <div
      class="ChoreComment__details"
    >
      <div
        class="ChoreComment__image"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular Profile__avatar--component css-tjtxwb-MuiAvatar-root"
        >
          <img
            alt="default-profile-pic.gif"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="default-profile-pic.gif"
          />
        </div>
      </div>
      <div
        class="ChoreComment__info"
      >
        <p
          class="ChoreComment__name"
        >
          testuser1
        </p>
        <p
          class="ChoreComment__date"
        >
          November 4, 2021 @ 10:24 PM
        </p>
        <div
          class="ChoreComment__comment"
        >
          <p>
            I'm commenting on the chore
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
